txpool/txpool_test.go

```diff
@@ -321,6 +342,23 @@ func TestAddTxErrors(t *testing.T) {
 		<-pool.promoteReqCh
 	})
 
+	t.Run("FillTxPoolToTheLimit - AddTx", func(t *testing.T) {
+		t.Parallel()
+
+		pool := setupPool()
+
+		// fill the pool leaving only 1 slot
+		pool.gauge.increase(defaultMaxSlots - 1)
+
+		// create tx requiring 1 slot
+		tx := newTx(defaultAddr, 0, 1, types.LegacyTxType)
+		tx = signTx(tx)
+
+		//	enqueue tx
+		assert.NoError(t, pool.AddTx(tx))
+		<-pool.promoteReqCh
```

Comment:
Does it make sense to assert the length of the tx pool at this point?

Reply:
It can be checked if there is a newly added transaction in the index map.